### ISSUE TITLE
Enable extension developer setting in Chrome automatically in dev mode

### ISSUE THREAD

**Issue by aklinker1** [wxt-dev/wxt#137]
### Feature Request

When running `wxt` to develop and extension, it's really annoying to have to enable developer mode every time to inspect the different JS contexts of your extension. We should be able to enable it automatically via CLI flags? I assume the switch is just a preference that we can control via CLI flags.

#### Is your feature request related to a bug?

https://github.com/wxt-dev/wxt/issues/135

### What are the alternatives?

N/A

### Additional context

Also a problem with Edge, so we should add the setting to edge and other chromium browsers as well. Hopefully they're all the same internal preference...

**Comment by aklinker1** [comment]
This doesn't seem possible. But you can enable it for all profiles, including new ones that WXT creates during dev mode:

https://stackoverflow.com/questions/32280949/enable-developer-mode-extension-for-chrome-browser-using-cmd#answer-32284345

**Comment by aklinker1** [comment]
I kept looking into this, and it is possible, but it's blocked by https://github.com/GoogleChrome/chrome-launcher/issues/323.

`web-ext` allows passing a custom `chromeLaunch` option, which I can use to apply custom prefs.

https://github.com/wxt-dev/wxt/blob/89d15babc2a4acf025dc1dfde14c2c32606a7e39/src/core/runners/web-ext.ts#L56

https://github.com/mozilla/web-ext/blob/a83f4c55a8fe0d2323d2d17ffbfa1754ec32f49c/src/extension-runners/chromium.js#L43

```ts
import { launch } from 'chrome-launcher';

await webExt.default.cmd.run({
  ...finalConfig,
  chromiumLaunch: (options) => launch({ ...options, prefs: { ... }
}, options); 
```

**Comment by aklinker1** [comment]
Can't pass the custom `chromiumLaunch` into the JS API... Need to wait for `web-ext` to add support, or add the feature to my fork. I'd rather contribute this to `web-ext` though.

https://github.com/mozilla/web-ext/issues/2899

**Comment by aklinker1** [comment]
This appears to be broken again

**Comment by aklinker1** [comment]
Need to figure out what the new preference is. Old preference: 

https://github.com/wxt-dev/web-ext-run/blob/a5d1e24b528911eb3b7d93439c45cc30094682be/src/extension-runners/chromium.js#L34-L36

Here's where we can set the new one if we find it:

https://github.com/wxt-dev/wxt/blob/90bb145d91f08faa13dae6ad17071ca36cb1441d/packages/wxt/src/core/runners/web-ext.ts#L96-L108

**Comment by hzuo** [comment]
@aklinker1 bumping into this issue repeatedly too! Did you end up figuring out what the new preference is? I'm ok with manually setting it via chromiumPrefs.

**Comment by aklinker1** [comment]
Nope, when I looked at the preferences file there was no flag that appeared to change when toggling the setting.

**Comment by gb-jos** [comment]
@hzuo @aklinker1 bumped into this issue trying to figure out where the pref is, seemingly it is in "Secure Preferences" file now instead of "Preferences"

**Comment by PatrykKuniczak** [comment]
That's really annoying, we need to fix it :)

**Comment by aklinker1** [comment]
> bumped into this issue trying to figure out where the pref is, seemingly it is in "Secure Preferences" file now instead of "Preferences"

Lovely. I won't get to this for a while, it's a change required on `web-ext`. Someone else is free to open a PR. But this has changed some much I closed my old PR for regular preferences.

You can refer to this PR for reference: https://github.com/mozilla/web-ext/pull/2912

**Comment by garretwilson** [comment]
I had been seeing issues when WXT tried to live-update the service worker: the extension would just disappear! I realized that it would become disabled because developer mode wasn't enabled. (I'm using Windows.)

After enabling [browser profile persistence](https://wxt.dev/guide/essentials/config/browser-startup.html#persist-data), each time I would use `npm run dev` I wouldn't even see the changes in the browser! I finally realized that this was the same issue: Chrome wasn't reloading the extension because it had gotten "stuck" because developer mode wasn't turned on, and this "stuck" state was now persisted.

Once I turned on developer mode manually, using [browser profile persistence](https://wxt.dev/guide/essentials/config/browser-startup.html#persist-data) seems to keep it on and everything is updating now.

In summary it would be nice if WXT would turn on developer mode automatically, but using [browser profile persistence](https://wxt.dev/guide/essentials/config/browser-startup.html#persist-data) is a workaround (still requiring manually to select developer mode once). Something about this might be helpful in the docs.

**Comment by PatrykKuniczak** [comment]
@aklinker1 We are able to solve this issue in some way?
Maybe you are able to finish your work on PR in [web-ext](https://github.com/mozilla/web-ext/pull/2912), now it has more sense to make it one more time, instead of rebase on this closed one.

But anyway are you even consider to do it, or you have other plan to give possibility to our users to enable this option?

**Comment by aklinker1** [comment]
I'm not going to continue this work. Chrome has changed where the value is stored multiple times now, breaking WXT each time. I'm not gonna fight against that 🤷 Gotta pick your battles.

If someone else wants to maintain this feature, feel free to continue where I left off on my [`web-ext` PR](https://github.com/mozilla/web-ext/pull/2912).

**Comment by PatrykKuniczak** [comment]
@aklinker1 It's released already:
https://github.com/mozilla/web-ext/releases/tag/10.2.0

But i can't bump it, because we're using https://github.com/wxt-dev/web-ext-run
But you have a plan to migrate from it, to `web-ext` directly, when you'll do this, this feature will be implemented.

**Comment by aklinker1** [comment]
Gonna do a major release soon, https://github.com/wxt-dev/wxt/pull/2447, will be fixed in that. We're adding `web-ext` as a peer, dropping my fork `web-ext-run`.

**Comment by PatrykKuniczak** [comment]
@aklinker1 Good to hear.
I've tried my best and use a part of your work, to make it possible ❤️
